compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -366,6 +366,12 @@ object desugar {
       (if (args.isEmpty) tycon else AppliedTypeTree(tycon, args))
         .withPos(cdef.pos.startPos)
 
+    def isHK(tparam: Tree): Boolean = tparam match {
+      case TypeDef(_, LambdaTypeTree(tparams, body)) => true
```

Comment:
I don't think that's enough in general, e.g. `enum Foo[F <: [X] => Any]`, but maybe we're ok with that since that's only a temporary solution.

Reply:
Yes, I think the right solution cannot be done at Desugar, it needs full type information.